src/parser/heredoc.c

```diff
@@ -0,0 +1,34 @@
+#include <minishell.h>
+
+// check !line later, now resolving a segfault
+// error code CNTRL_D = -2 could be a solution
+int32_t	here_doc(char *delimiter)
+{
+	int		fd[2];
+	char	*line;
+
+	if (!delimiter || ft_strlen(delimiter) == 0)
+		return (ERROR);
+	if (pipe(fd) == -1)
+		return (ERROR);
+	while (1)
+	{
+		line = readline("> ");
+		if (!line)
+		{
+			close(fd[0]);
+			close(fd[1]);
+			return (ERROR);
+		}
+		if (ft_strncmp(line, delimiter, ft_strlen(delimiter)) == 0)
```

Comment:
ft_strlen(delimiter) should be plus one to also check if the character after the last index of the delimiter is also a '\0' character